src/Pages/ProductListPage.js

```diff
@@ -38,67 +43,82 @@ const ProductList = () => {
         });
     }
 
-    const handleCheckboxChange = (event) =>{
+
+    const handleCheckboxChange = (event) => {
         const arrayOfchosedCategories = chosedCategories.items;
 
         const isItOnTheArray = arrayOfchosedCategories.some(
             category => category === event.target.name
         );
 
-        if(!isItOnTheArray)
+        if(!isItOnTheArray && event.target.checked)
             addCategory(arrayOfchosedCategories, event.target.name);
-        if(isItOnTheArray)
+        if(isItOnTheArray && !event.target.checked)
             removeCategory(arrayOfchosedCategories, event.target.name);
 
+        if(chosedCategories.items.length > 0)
+            setShowResetButton(true)
+
         const filteredProducts = filter();
```

Comment:
Didn't noticed this the first time I reviewed this. `filteredProducts` is opaquing the value from line 17. Change the name to avoid this.